src/fs/fs.c

```diff
@@ -1676,6 +1676,32 @@ DEFINE_SYSCALL(umask, int, mask)
   return vfs_umask(mask);
 }
 
+DEFINE_SYSCALL(mknodat, int, dirfd, gaddr_t, path_ptr, l_mode_t, mode, l_dev_t, dev) {
+  char name[LINUX_PATH_MAX];
+  strncpy_from_user(name, path_ptr, sizeof name);
```

Comment:
@kazumi007 

Can you add a guard here that checks if strncpy_from_user succeeded?